habitat_baselines/rl/ddppo/data_generation/create_gibson_large_dataset.py

```diff
@@ -0,0 +1,87 @@
+"""This script is used to generate customized Gibson training splits for the
+PointNav task. The scenes in Gibson are ranked from 1-5 based on the
+reconstruction quality (see https://arxiv.org/pdf/1904.01201.pdf for more
+details). This script generates data for all scenes with a minimum quality of
+q_thresh.
+"""
+import glob
+import gzip
+import json
+import multiprocessing
+import os
+import os.path as osp
+
+import tqdm
+
+import habitat
+import habitat_sim
+from habitat.datasets.pointnav.pointnav_generator import (
+    generate_pointnav_episode,
+)
+
+num_episodes_per_scene = int(1e4)
+
+
+def safe_mkdir(path):
+    try:
+        os.mkdir(path)
+    except:
+        pass
+
+
+def _generate_fn(scene):
+    cfg = habitat.get_config()
+    cfg.defrost()
+    cfg.SIMULATOR.SCENE = scene
+    cfg.SIMULATOR.AGENT_0.SENSORS = []
+    cfg.freeze()
+
+    sim = habitat.sims.make_sim("Sim-v0", config=cfg.SIMULATOR)
+
+    dset = habitat.datasets.make_dataset("PointNav-v1")
+    dset.episodes = list(
+        generate_pointnav_episode(
+            sim, num_episodes_per_scene, is_gen_shortest_path=False
+        )
+    )
+    for ep in dset.episodes:
+        ep.scene_id = ep.scene_id[len("./data/scene_datasets/") :]
+
+    scene_key = scene.split("/")[-1].split(".")[0]
+    out_file = (
+        f"./data/datasets/pointnav/gibson/v2/train_large/content/"
+        f"{scene_key}.json.gz"
+    )
+    os.makedirs(osp.dirname(out_file), exist_ok=True)
+    with gzip.open(out_file, "wt") as f:
+        f.write(dset.to_json())
+
+
+# Load train / val statistics
+dataset_statistics = json.load(
```

Comment:
Can we make it as separate function and call it from `main` method. That will make easier to test it and re-use it.

Reply:
Got it. I just need to put the code from hereon into a `generate_gibson_large_dataset()` function and call that from an `if __name_ == "__main__":` block, right?